src/test/java/io/cdap/plugin/salesforce/connector/SalesforceConnectorTest.java

```diff
@@ -96,6 +98,33 @@ private Schema getPluginSchema() throws IOException {
       "\"null\"]},{\"name\":\"userId\",\"type\":\"string\"}]}";
 
     return Schema.parseJson(schemaString);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testSinkValidateSchema() throws IOException, ConnectionException {
+    SalesforceSinkConfig config = new SalesforceSinkConfig("REFERENCE_NAME",
+                                                           "CONSUMER_KEY", "CONSUMER_SECRET",
+                                                           "USERNAME", "PASSWORD",
+                                                           "LOGIN_URL", 30000, "sObject", "Insert", null,
+                                                           ConcurrencyMode.Parallel.name(), "1000000", "10000",
+                                                           "Fail on Error",
+                                                           "SECURITY_TOKEN",
+                                                           null, null, true);
 
+    if (config.isDatatypeValidation()) {
```

Comment:
The test should not call isDatatypeValidation() directly - call the validate method and verify in the test that input schema was validated.
Similarly, add another test for the case where datatypeValidation=false and verify in the test that we don't do schema validation

Reply:
I have tried that but the checkCompatibility method inside validateInputSchema  is returning void and its class is mocked static, so we need to upgrade Mockito library for that and upgrading that causing various test failures in existing test cases. As this is a P1 issue, we tried to expedite it, and that will require more time. Can we add a ETL test case for time being to test it fully.

Reply:
Sure, we can update tests later. You can create a Jira for tracking purposes and update the tests in another PR